android-lib/src/androidTest/java/com/kinvey/androidTest/request/RequestTest.java

```diff
@@ -0,0 +1,151 @@
+    private TestManager<Person> testManager;
+    private DataStore<Person> store;
+
+    @Before
+    public void setUp() throws InterruptedException, IOException {
+        Context mMockContext = new RenamingDelegatingContext(InstrumentationRegistry.getInstrumentation().getTargetContext(), "test_");
+        client = new Client.Builder(mMockContext).build();
+        testManager = new TestManager<>();
+        testManager.login(USERNAME, PASSWORD, client);
+
+    }
+
+    @After
+    public void tearDown() {
+        client.performLockDown();
+        if (client.getKinveyHandlerThread() != null) {
+            try {
+                client.stopKinveyHandlerThread();
+            } catch (Throwable throwable) {
+                throwable.printStackTrace();
+            }
+        }
+    }
+
+
+    @Test
+    public void testDefaultTimeOutValue() throws InterruptedException, IOException {
+        store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        testManager.cleanBackend(store, StoreType.SYNC);
+        client.enableDebugLogging();
+        assertEquals(DEFAULT_TIMEOUT, client.getRequestTimeout());
+
+        Person person = new Person();
+        person.setUsername(TEST_USERNAME);
+        DefaultKinveyClientCallback callback = testManager.save(store, person);
+        assertNotNull(callback);
+        assertNotNull(callback.getResult());
+        assertNull(callback.getError());
+        assertNotNull(callback.getResult().getUsername());
+        assertEquals(TEST_USERNAME, callback.getResult().getUsername());
+
+        DefaultKinveyPushCallback pushCallback = testManager.push(store);
+        assertNotNull(pushCallback);
+        assertNotNull(pushCallback.getResult());
+        assertNull(pushCallback.getError());
+        assertEquals(1, pushCallback.getResult().getSuccessCount());
+    }
+
+    @Test
+    public void testSetCustomTimeoutAndCheckException() throws InterruptedException, IOException {
+        store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        testManager.cleanBackend(store, StoreType.SYNC);
+        assertEquals(DEFAULT_TIMEOUT, client.getRequestTimeout());
+
+        client.setRequestTimeout(1);
+        assertEquals(1, client.getRequestTimeout());
+
+        Person person = new Person();
+        person.setUsername(TEST_USERNAME);
+        DefaultKinveyClientCallback callback = testManager.save(store, person);
+        assertNotNull(callback);
+        assertNotNull(callback.getResult());
+        assertNull(callback.getError());
+        assertNotNull(callback.getResult().getUsername());
+        assertEquals(TEST_USERNAME, callback.getResult().getUsername());
+
+        DefaultKinveyPushCallback pushCallback = testManager.push(store);
+        assertNotNull(pushCallback);
+        assertNotNull(pushCallback.getError());
+        assertEquals("SocketTimeoutException", pushCallback.getError().getClass().getSimpleName());
```

Comment:
Should we assertNull(pushCallback.getResult()) here?

Reply:
In general if `pushCallback` has error it shouldn't have any result in callback. But I think it's good to have a double check on this logic. `assertNull(pushCallback.getResult())` was added.